### DISCUSSION TITLE
Tutorial on spacing in block themes

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#68] [Topic Ideas]
Currently, we have some pretty good [docs for spacing](https://developer.wordpress.org/themes/advanced-topics/theme-json/#spacing) in the theme handbook, which serves as a solid foundation on how to implement spacing in block themes.  However, like most docs, the sheer number of options presented doesn't present a look at how to solve each problem that developers might come across.

On a recent outreach call, for example, I had a developer who was having trouble with spacing on specific blocks being overwritten by the global `blockGap`.  This led them to disabling `blockGap` altogether and implementing CSS-based spacing instead.  The goal is to move them entirely over to `theme.json` if possible.  These are the sort of real-world problems that pop up and should be explored in a more tutorial-like piece of content.

So, I'm opening this up for discussion.  Perhaps there are other spacing-specific things that need to be covered for developers??

I envision this post covering all of the existing spacing options (and when to use them) for theme developers and also covering any gotchas, tips, and tricks.

**Comment by justintadlock** [comment]
Just noting that I'll pick this up, assuming it's approved in the editorial meeting this week.

**Comment by bph** [comment]
Discussion is now locked. Please add further comments to the issue: 

- #88

**Comment by bph** [comment]
published: [Everything you need to know about spacing in block themes](https://developer.wordpress.org/news/2023/03/everything-you-need-to-know-about-spacing-in-block-themes/)
